### PR TITLE
add uuid support.

### PR DESCRIPTION
**Pull request by shobeiry**
add uuid support.

### CODE REVIEW

**Comment by frital** [comment]
Hi @shobeiry !
I'm also need this, and would to start to work on a PR for one of the most popular role / permissions laravels packages (maybe this or laratrust). But in any case, in you PR, i think that a "static::creating" boot method is missed, or not? Because since you lost the auto-increment on db level, you will need generate and assign an uuid when a new model (role or permission) is created. Pls correct me if i am wrong.
Thx in advance!!

**Review by shobeiry — COMMENTED** [review]
now auto generate uuid on create method added to models.

File: src/Models/Role.php [hunk 1]
```diff
@@ -21,11 +21,32 @@ class Role extends Model implements RoleContract
...
         parent::__construct($attributes);
     }
 
+    public static function boot()
+    {
+        parent::boot();
+        if (config('permission.models.keys_type') == 'uuid') {
+            self::creating(function ($model) {
+                if (is_null($model->id)) {
+                    $model->id = Str::uuid()->toString();
```
Comment (drbyte) [inline src/Models/Role.php:44]: Should `Str` be imported?

**Review by shobeiry — COMMENTED** [review]
i fix Str not importing in Role model.

**Comment by freekmurze** [comment]
Hi, thanks for this. I'm going to pass on this for now, but I'll keep this feature in mind when working on the next major release of the package.

**Pull request closed without merging**